proto/cluster.proto

```diff
@@ -0,0 +1,78 @@
+syntax = "proto3";
+
+option go_package = "github.com/ray-project/kuberay/proto/go_client";
+package proto;
+
+import "google/protobuf/timestamp.proto";
+
+
+message Cluster {
+  // Required input field. Unique Cluster name provided by user.
+  string name = 1;
+
+  // Required input field. Cluster's namespace provided by user
+  string namespace = 2;
+
+  // Required field. This field indicates the user who owns the cluster.
+  string user = 3;
+
+  // Optional input field. Ray cluster version
+  string version = 4;
+
+  // Optional field.
+  enum Environment {
+    DEV = 0;
+    TESTING = 1;
+    STAGING = 2;
+    PRODUCTION = 3;
+  }
+  Environment environment = 5;
+
+  // Required field. This field will be used to retrieve right ray container
+  string image = 6;
+  
+  // Required field. This field indicates ray cluster configuration
+  ClusterSpec cluster_spec = 7;
```

Comment:
1. Do we want to make ClusterSpec reusable? (if so, we need to persist spec object)

2. Do we want to embed the object or use reference like "cluster_spec_name"? 

- embed the object -> how can user figure out the the right object? What if they like to reuse the object? What if they copy from cluster spec but make minor modifications, if that case, we can not find the object? 

- name -> This is reliable but does need user to have multi steps to create the cluster

Reply:
I think it is a little difficult to reuse ClusterSpec in multi clusters. Most clusters have different spec.

Reply:
If we do not like to reuse it, that's easier and we can just embed the object

Reply:
I remove `clusterSpec.name`. Everytime, user should pass entire clusterSpec object into the POST payload